src/control/lib/ipmctl/nvm_types.go

```diff
@@ -50,17 +70,26 @@ type DeviceDiscovery struct {
 	Manufacturing_location   uint8
 	Manufacturing_date       uint16
 	Part_number              [21]int8
-	Fw_revision              [25]int8
-	Fw_api_version           [25]int8
+	Fw_revision              Version
+	Fw_api_version           Version
 	Pad_cgo_2                [5]byte
 	Capacity                 uint64
 	Interface_format_codes   [9]uint16
 	Security_capabilities    _Ctype_struct_device_security_capabilities
 	Device_capabilities      _Ctype_struct_device_capabilities
-	Uid                      [22]int8
+	Uid                      DeviceUID
 	Lock_state               uint32
 	Manageability            uint32
 	Controller_revision_id   uint16
 	Reserved                 [48]uint8
 	Pad_cgo_3                [6]byte
 }
+
+// DeviceFirmwareInfo represents an ipmctl device_fw_info structure
+type DeviceFirmwareInfo struct {
+	ActiveFWVersion Version
+	StagedFWVersion Version
+	FWImageMaxSize  uint32
+	FWUpdateStatus  uint32
```

Comment:
Is the set of status values known? If so, might be good to put them in a const list.

Reply:
Adding the enum